File: examples/user-tasks/src/main/java/io/littlehorse/examples/EmailSender.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHTaskMethod;
import io.littlehorse.sdk.worker.WorkerContext;

public class EmailSender {

    @LHTaskMethod("send-email")
    public void sendEmail(
        String address,
        String content,
        WorkerContext workerContext
    ) {
        if (workerContext.getUser() != null) {
            System.out.println(
                "Received variable by " + workerContext.getUser().getId()
            );
        } else if (workerContext.getUserGroup() != null) {
            System.out.println(
                "Received variable by " + workerContext.getUserGroup().getId()
            );
        }

        System.out.println("\n\nSending email to " + address);
        System.out.println("Content: " + content);
    }
}

File: sdk-dotnet/Examples/UserTasksExample/EmailSender.cs
using LittleHorse.Sdk.Worker;

namespace UserTasksExample;

public class EmailSender
{
    [LHTaskMethod("send-email")]
    public async Task SendEmail(string address, string content, LHWorkerContext workerContext)
    {
        if (workerContext.UserId != null)
        {
            Console.WriteLine($"Received variable by {workerContext.UserId}");
        }
        else if (workerContext.UserGroup != null)
        {
            Console.WriteLine($"Received variable by {workerContext.UserGroup}");
        }

        Console.WriteLine($"Sending email to {address}");
        Console.WriteLine($"Content: {content}");
    }
}